integrations/jira/main.py

```diff
@@ -23,63 +35,59 @@ async def setup_application() -> None:
         )
         return
 
-    jira_client = JiraClient(
-        logic_settings["jira_host"],
-        logic_settings["atlassian_user_email"],
-        logic_settings["atlassian_user_token"],
-    )
-
-    await jira_client.create_events_webhook(
-        logic_settings["app_host"],
-    )
+    client = create_jira_client()
+    await client.create_events_webhook(app_host)
 
 
 @ocean.on_resync(ObjectKind.PROJECT)
 async def on_resync_projects(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
-    client = JiraClient(
-        ocean.integration_config["jira_host"],
-        ocean.integration_config["atlassian_user_email"],
-        ocean.integration_config["atlassian_user_token"],
-    )
-
+    client = create_jira_client()
     async for projects in client.get_paginated_projects():
-        logger.info(f"Received project batch with {len(projects)} issues")
+        logger.info(f"Received project batch with {len(projects)} projects")
         yield projects
 
 
 @ocean.on_resync(ObjectKind.ISSUE)
 async def on_resync_issues(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
-    client = JiraClient(
-        ocean.integration_config["jira_host"],
-        ocean.integration_config["atlassian_user_email"],
-        ocean.integration_config["atlassian_user_token"],
-    )
+    client = create_jira_client()
 
-    async for issues in client.get_paginated_issues():
+    config = cast(JiraResourceConfig, event.resource_config)
+    jql = config.selector.jql if config else None
+
+    async for issues in client.get_paginated_issues(jql):
         logger.info(f"Received issue batch with {len(issues)} issues")
         yield issues
 
 
+@ocean.on_resync(ObjectKind.TEAM)
+async def on_resync_teams(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    client = create_jira_client()
+    org_id = ocean.integration_config.get("atlassian_organization_id")
+
+    if not org_id:
+        logger.info("Skipping team sync - no organization ID configured")
+        return
+
+    selector = cast(TeamResourceConfig, event.resource_config).selector
+    async for teams in client.get_paginated_teams(org_id):
+        logger.info(f"Received team batch with {len(teams)} teams")
+        if selector.include_members:
```

Comment:
Great suggestion.

Reply:
Although this would simplify the API, I feel having separate methods provides clearer boundaries and makes the code more maintainable. Each method has a distinct purpose - fetching teams, fetching members, and enrichment,

Reply:
I agree that separate methods make it better. You can have separate methods in the client class still but we expose one to the integration to use. When testing, you can always check if the enrich method is called or if the `__members` key is included in the final data.